File: SDKs/amazonaws.com/lightsail/2016-11-28/java/lib/src/main/java/openapisdk/models/shared/ContainerServiceStateDetail.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * ContainerServiceStateDetail
 * Describes the current state of a container service.
**/
public class ContainerServiceStateDetail {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("code")
    public ContainerServiceStateDetailCodeEnum code;
    public ContainerServiceStateDetail withCode(ContainerServiceStateDetailCodeEnum code) {
        this.code = code;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("message")
    public String message;
    public ContainerServiceStateDetail withMessage(String message) {
        this.message = message;
        return this;
    }
}

File: SDKs/amazonaws.com/lightsail/2016-11-28/python/src/sdk/models/shared/containerservicestatedetail.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import containerservicestatedetailcode_enum as shared_containerservicestatedetailcode_enum


@dataclass_json
@dataclasses.dataclass
class ContainerServiceStateDetail:
    r"""ContainerServiceStateDetail
    Describes the current state of a container service.
    """
    
    code: Optional[shared_containerservicestatedetailcode_enum.ContainerServiceStateDetailCodeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('code') }})
    message: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('message') }})
    
